contracts/storage/PeripheryStorage.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity =0.8.9;
+
+import "../interfaces/IERC20Minimal.sol";
+import "../interfaces/fcms/IFCM.sol";
+import "../interfaces/IMarginEngine.sol";
+import "../interfaces/IVAMM.sol";
+
+contract PeripheryStorageV1 {
+    // Any variables that would implicitly implement an IMarginEngine function if public, must instead
+    // be internal due to limitations in the solidity compiler (as of 0.8.12)
+
+    /// @dev Wrapped ETH interface
+    IWETH public _weth;
+
+    /// @dev Voltz Protocol vamm => LP Margin Cap in Underlying Tokens
+    /// @dev LP margin cap of zero implies no margin cap
+    mapping(IVAMM => int256) public _lpMarginCaps;
```

Comment:
I can see this comment for other Storage contracts. Not sure what it means
// Any variables that would implicitly implement an IMarginEngine function if public, must instead
// be internal due to limitations in the solidity compiler (as of 0.8.12)

Reply:
to be safe i suggest following the same pattern as we do with VAMM and Margin Engine, will be helpful if you can elaborate on the comments @ioanaroceanu2000 highlighted above and we can add them to the comments as well to avoid confusion in the future

Reply:
We are talking about state variables that are also part of the interface declared in `IPeriphery`.

I would be happy with such variables being public, with the same name as the mandatory `IPeriphery` function (in this case that's `lpMarginCaps()` and not `_lpMarginCaps()`) - except that I couldn't get that to compile. Solidity's compilation of `Periphery` was not being smart enough to understand that the function `lpMarginCaps()` was implemented implicitly in a superclass (`PeripheryStorage`).

(My actual experience was with MarginEngine/VAMM, but example above uses the periphery for clarity.)

So I think that, to satisfy the compiler, we have to make these sorts of things internal and write an explicit getter, which you have already done here: https://github.com/Voltz-Protocol/voltz-core/pull/168/files#diff-0b5c6a3e66327494bc1e399dcbc8d4fc25f35e9b078d86080cd9968adfdf8c5fR58-R61

And if we do have an explicit getter for any reason, as we have here, then the underlying data structure / variable should certainly be internal so that we don't end up with two getters.

Does this make sense, @ioanaroceanu2000 & @arturbeg ? If so, perhaps you can would be kind enough to clarify the code comments as you see fit?

Thanks!

Reply:
got it, thanks for the clarification @CyclopsRex

Reply:
makes sense now, I'll add the changes